File: app/javascript/channels/building_channel.js
import consumer from "channels/consumer"

const building = document.getElementById('building').dataset.buildingId;


consumer.subscriptions.create({channel: "BuildingChannel", building_id: building}, {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log("Connected to BuildingChannel with building_id: " + building);
  },

  disconnected() {
    // Called when the subscription has been terminated by the server

  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    console.log("Received data on BuildingChannel: ", data);
  }
});

File: app/channels/building_channel.rb
class BuildingChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    stream_from "building_#{params[:building_id]}_channel"

    RoomOccupancyStatusJob.perform_later(params[:building_id])
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end
